TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Excursion.java

```diff
@@ -162,19 +154,19 @@ public void setExcursionDuration(Duration excursionDuration) {
     }
 
     /**
-     * Retrieves all trips for this excursion
-     * @return set of trips
+     * Retrieves all trip for this excursion
+     * @return set of trip
      */
-    public Set<Trip> getTrips() {
-        return trips;
+    public Trip getTrip() {
+        return trip;
     }
 
     /**
-     * Sets trips to this excursion entity
-     * @param trips set of {@link Trip}s
+     * Sets trip to this excursion entity
+     * @param trip set of Trip
      */
-    public void setTrips(Set<Trip> trips) {
-        this.trips = trips;
+    public void setTrip(Trip trip) {
+        this.trip = trip;
```

Comment:
I suggest adding trip.addExcursion(this); to keep relationship synced.

Reply:
only adding no need to removing?
and do you mean like this?

public void addTrip(Trip trip) {
        trip.addExcursion(this);
    }